src/apps/intel_mp/intel_mp.lua

```diff
@@ -446,9 +447,31 @@ function Intel:new (conf)
          alarm_text='Ingress packet-rate exceeds 2MPPS which can cause packet drops.'
       }
    }
-   self.ingress_packet_rate_alarm = CallbackAlarm.new(ingress_packet_rate,
+   self.ingress_packet_rate_alarm = CallbackRateAlarm.new(ingress_packet_rate,
       1, 2e6, function() return self:rxpackets() end)
 
+
+   if self.master then
+      alarms.add_to_inventory {
+         [{alarm_type_id='phy-down'}] = {
+            resource=self.pciaddress,
+            has_clear=true,
+            description='Interface is down',
+         }
+      }
+      local phy_down_state = alarms.declare_alarm {
+         [{resource=self.pciaddress,alarm_type_id='phy-down'}] = {
+            perceived_severity='major',
+            alarm_text='Interface is down',
+         }
+      }
+      self.phy_down_state_alarm = CallbackAlarm.new(phy_down_state,
+         1, function()
+            -- Alarm is not active when link is active
+            return not self:link_status()
+         end)
+   end
+
```

Comment:
I think you should be able to code this behavior without implementing a new `CallbackAlarm` for boolean values. I explain how.

Current `CallbackAlarm` takes 4 parameters: `alarm`, `period`, `limit` and a `callback`. Argument `limit` is the value that determines whether the alarm should be raised or cleared. So if the value of the callback is a boolean you can convert the result to an integer (1 or 0) and set `limit` as 0. Something like:

```lua
self.phy_down_state_alarm = CallbackAlarm.new(phy_down_state,
   1, 0, function()
   -- Alarm is not active when link is active
   if not self:link_status() return 1 else return 0 end
end)
```
So when the value returned is 1, since it's higher than `limit` the alarm is raised. The alarm is cleared otherwise.

Reply:
Yep, fair point - I'll give that a shot

Reply:
Hi @dpino I remember why this doesn't work now - the code in `Alarm:check()` calculates a difference (rate) based on `value - last_value` - this means that the alarm will automatically resolve itself the second time it runs (as 1 - 1 = 0 and 0 - 0 = 0) if the state has not changed.

Obviously I could use `self.alarm:raise()` and `:clear()` manually without using the Alarm classes but that seems messy - thoughts?